dataprep/clean/clean_phone.py

```diff
@@ -20,15 +20,45 @@
     [-. (]*
     (?P<area>\d{3})?
     [-. )\/]*
-    (?P<office>\d{3})
+    (?:(?P<office>\d{3})
     [-. \/]*
-    (?P<station>\d{4})
+    (?P<station>\d{4})|
+    (?P<letters>[0-9A-Z-. \/]{7,13}))
```

Comment:
How did you choose 13 to be the upper bound here?

Reply:
13 allows for one delimiter between each letter, e.g. "1-800-G-O-T-J-U-N-K", which is sort of an extreme case (splitting each letter on its own). I also considered using a tighter bound like 9 since it's hard to form more than 3 words using 7 characters (e.g. "1-800-4-MY-XBOX").

Reply:
That makes sense, thanks for the explanation. What do you think about adding the non-greedy qualifier `?` like `{7,13}?` so that matching the phone extension is prioritized. I think in our current regex it's okay, but in the future, if we want to support extensions with all capital letters like "1-800-4-MY-XBOX EXT 1" or "1-800-4-MY-XBOX E 1" the `letters` capturing group won't match the extension.

Also could you please add the example "1-800-G-O-T-J-U-N-K" in the unit test?

Reply:
Sure, will do.